tests/test_encoding.py

```diff
@@ -88,6 +89,300 @@ def test_encode_coded_const_infer_order(self) -> None:
+            base_data_type=DataType.A_INT32,
+            base_type_encoding=Encoding.SM,
+            is_highlow_byte_order=True)
+        self.assertEqual(decoded, -0x1234)
+
+    def test_float_encodings(self) -> None:
+        # FLOAT32
+        encode_state = EncodeState()
+        with self.assertRaises(OdxError):
+            encode_state.emplace_atomic_value(
+                internal_value=-1.234,
+                bit_length=24,
+                base_data_type=DataType.A_FLOAT32,
+                base_type_encoding=Encoding.NONE,
+                is_highlow_byte_order=True,
+                used_mask=None)
+        encode_state.emplace_atomic_value(
+            internal_value=-1.234,
+            bit_length=32,
+            base_data_type=DataType.A_FLOAT32,
+            base_type_encoding=Encoding.NONE,
+            is_highlow_byte_order=True,
+            used_mask=None)
+        self.assertEqual(encode_state.coded_message, b'\xb6\xf3\x9d\xbf'[::-1])
+
+        decode_state = DecodeState(encode_state.coded_message)
+        with self.assertRaises(OdxError):
+            decoded = decode_state.extract_atomic_value(
+                bit_length=24,
+                base_data_type=DataType.A_FLOAT32,
+                base_type_encoding=Encoding.NONE,
+                is_highlow_byte_order=True)
+        decoded = decode_state.extract_atomic_value(
+            bit_length=32,
+            base_data_type=DataType.A_FLOAT32,
+            base_type_encoding=Encoding.NONE,
+            is_highlow_byte_order=True)
+        self.assertTrue(abs(float(decoded) - (-1.234)) < 1e-6)
+
+        # FLOAT64
+        encode_state = EncodeState()
+        with self.assertRaises(OdxError):
+            encode_state.emplace_atomic_value(
+                internal_value=-1.234,
+                bit_length=24,
+                base_data_type=DataType.A_FLOAT64,
+                base_type_encoding=Encoding.NONE,
+                is_highlow_byte_order=True,
+                used_mask=None)
+        encode_state.emplace_atomic_value(
+            internal_value=-1.234,
+            bit_length=64,
+            base_data_type=DataType.A_FLOAT64,
+            base_type_encoding=Encoding.NONE,
+            is_highlow_byte_order=True,
+            used_mask=None)
+        self.assertEqual(encode_state.coded_message, b'X9\xb4\xc8v\xbe\xf3\xbf'[::-1])
+
+        decode_state = DecodeState(encode_state.coded_message)
+        with self.assertRaises(OdxError):
+            decoded = decode_state.extract_atomic_value(
+                bit_length=24,
+                base_data_type=DataType.A_FLOAT64,
+                base_type_encoding=Encoding.NONE,
+                is_highlow_byte_order=True)
+        decoded = decode_state.extract_atomic_value(
+            bit_length=64,
+            base_data_type=DataType.A_FLOAT64,
+            base_type_encoding=Encoding.NONE,
+            is_highlow_byte_order=True)
+        self.assertTrue(abs(float(decoded) - (-1.234)) < 1e-9)
```

Comment:
What's this?

Reply:
this is a potential rounding error. (I don't want to rely on the fact that python uses 64 bit floating point values internally...)

Reply:
put that info as code comment